docs/troubleshooting.md

```diff
@@ -0,0 +1,141 @@
+NAME                               CATALOG             AGE
+...
+portworx                           My Catalog Source   14m
+postgres-operator                  My Catalog Source   14m
+postgresql                         My Catalog Source   14m
+postgresql-operator-dev4devs-com   My Catalog Source   14m
+prometheus                         My Catalog Source   14m
+...
+```
+
+If the opreator is not listed, rebuild the CatalogSource with the [operator-sdk bundle command](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#operator-bundle-commands).
+
+If the operator is present, check if the version you want is available:
+
+`$ kubectl -n my-namespace get packagemanifests my-operator -o yaml`
+
+### My CatalogSource cannot pull images from a private registry
+
+If you are attempt to pull images from a private registry, make sure to specify a secret key in the `CatalogSource.Spec.Secrets` field.
+
+## Subscription Troubleshooting
+
+This section assumes that you have a working `CatalogSource`.
+
+### How to debug a failing Subscription
+
+The Catalog operator will constantly update the Status of `Subscription` to reflect its current state. You can check the status of your `Subscription` with the following command:
+
+`$ kubectl -n my-namespace get subscriptions my-subscription -o yaml | yq r - status`
+
+>Note: It is possible that the `Status` is missing, which suggests that the Catalog operator is encountering an issue when processing the `Subscription` in a very early stage.
+
+If the status block does not provide enough information, check the [Catalog operator's logs](#how-to-view-the-catalog-operator-logs).
+
+### A subscription in namespace X can't install operators from a CatalogSource in namespace Y
+
+`Subscriptions` cannot install operators provided by `CatalogSources` that are not in the same namespace unless the `CatalogSource` is created in the `olm` namespace.
+
+## ClusterServiceVersion Troubleshooting
+
+### How to debug a failing CSV
+
+If the OLM operator encounters an unrecoverable error when attempting to install the operator the `CSV` will be placed in the `failed` phase. The `OLM` operator will constantly update the Status with useful information regarding the state of the `CSV`. You can check the status of your `CSV` with the following command:
+
+`$ kubectl -n my-catalogsource-namespace get csv prometheusoperator.0.32.0 -o yaml | yq r - status`
+
+>Note: It is possible that the Status is missing, which suggests that the OLM operator is encountering an issue when processing the `CSV` in a very early stage. You should respond by reviewing the logs of the OLM operator.
+
+You should typically pay special attention to the information within the `status.reason` and `status.message` fields. Please look in the [#failed-csv-reasons]
+
+If the status block does not provide enough information, check the [OLM operator's logs](#how-to-view-the-olm-operator-logs).
+
+#### Failed CSV Reasons
```

Comment:
+1 to this section. we should try to do this everywhere, this enumeration is really awesome.